hop/subscribe.py

```diff
@@ -72,12 +105,16 @@ def _main(args=None):
     # set offset
     start_offset = "earliest" if args.earliest else "latest"
 
+    # set timeout
+    t = args.timeout
```

Comment:
I'm a little wary of the way this code is shaping up. As it starts to use attributes on the `args` object, it gets harder to read. To know what `args.timeout` is, you have to read back upwards to get to `args = parser.parse_args`... but you also have to be prepared for `args` to be passed in to `_main` directly, and cross your fingers that it's an object with the right attributes. Meanwhile, they're also `config` which loads more configuration.

I think this PR represents the last limits before this really gets too unwieldy to keep working with it. We should probably refactor how configuration loading works before going any further - that can be a follow-on ticket.

Reply:
Sure, we can look into this more during a refactor. I based this setup on the [cli changes](https://github.com/scimma/hop-client/pull/60/files#diff-3a67069fe9c758c5cdc533b4d7ac90b3) from @myNameIsPatrick but it'd be good to consider the downstream effects of this.

Reply:
@spenczar, I can hopefully give some context into some of this.

So this whole business with `_main` and handling the `args` case is probably not needed, it's a pattern I've normally followed when building a CLI with a subparser so that subcommands are registered correctly, via https://github.com/scimma/hop-client/blob/feature/subscribe_notices/hop/__main__.py#L43. With some work, it can probably be simplified.

The rationale behind the `config` handling was to provide a way to do auth before some of the auth functionality was added to `adc-streaming`, so the goal there was to mimic how config handling is done via kafkacat (the -F and -X opts) to pass in directly to the stream instance. I would prefer some simplification of this and to provide a simpler auth interface and some defaults so that it "does the right thing" connecting to the hopskotch broker, and axe the custom config stuff entirely. That may or may not be doable, but having flexibility like this comes with additional complexity.

We should figure out what it is we ideally want to expose and then refactor our CLI as needed to provide a simple, consistent interface that does what we want. There's definitely some technical debt we'd need to pay off as we've added in features one by one over the sprints.